pylint/checkers/design_analysis.py

```diff
@@ -440,11 +444,22 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
                 args=(nb_parents, self.linter.config.max_parents),
             )
 
-        if len(node.instance_attrs) > self.linter.config.max_attributes:
+        # Something at inference time is modifying instance_attrs to add
+        # properties from parent classes. Given how much we cache inference
+        # results, mutating instance_attrs can become a real mess. Filter
+        # them out here until the root cause is solved.
+        # https://github.com/pylint-dev/astroid/issues/2273
```

Comment:
Should this be a release blocker?

Reply:
I doubt it, since the current situation is better (more deterministic).

Reply:
The primer results are looking a lot better.

Reply:
What's with the `deprecated-typing-alias` messages?

Reply:
It looks like the junk from #8790 is finally going away!

Reply:
Btw, I made a stab at the root cause in astroid and made a bit of progress, but there is still a way to go before mergeable.